src/data/nat/modeq.lean

```diff
@@ -389,3 +389,96 @@ lemma rotate_one_eq_self_iff_eq_repeat [nonempty α] {l : list α} :
     λ h, rotate_eq_self_iff_eq_repeat.mpr h 1⟩
 
 end list
+
+--------------------------------------------------------------------------------------------------
+--------------------------------------------------------------------------------------------------
+--  PR THIS vvvv  --------------------------------------------------------------------------------
+--------------------------------------------------------------------------------------------------
+--------------------------------------------------------------------------------------------------
+
+namespace nat
+
+variables {a b c d m n : ℕ}
+
+-- Apostol, Theorem 5.3
+lemma mul_left_iff (c : ℕ) (hc : 0 < c) : a ≡ b [MOD m] ↔ c * a ≡ c * b [MOD c * m] :=
+begin
+  have hc' : (c:ℤ) ≠ 0 := by simp [hc.ne.symm],
+  simp only [modeq_iff_dvd, int.coe_nat_mul, ←mul_sub, mul_dvd_mul_iff_left hc'],
+end
+
+lemma mul_right_iff (hc : 0 < c) : a ≡ b [MOD m] ↔ a * c ≡ b * c [MOD m * c] :=
+begin
+  have hc' : (c:ℤ) ≠ 0 := by simp [hc.ne.symm],
+  simp only [modeq_iff_dvd, int.coe_nat_mul, ←mul_sub_right_distrib, mul_dvd_mul_iff_right hc'],
+end
+
+
+
+-- Apostol, Theorem 5.5
+
+lemma Apostol_5_5' (h : a ≡ b [MOD m]) (hdm : d ∣ m) :
```

Comment:
Is this lemma really useful, though? I feel like the proof of this result is easier to come up with than the result itself.

Reply:
Yeah, I think you're right.  I've deleted this and inlined it into `dvd_iff_of_modeq_of_dvd`.